app/src/org/commcare/activities/FormEntryActivityUIController.java

```diff
@@ -493,20 +494,23 @@ private void createRepeatDialog() {
         final boolean backExitsForm = !details.relevantBeforeCurrentScreen;
         final boolean nextExitsForm = details.relevantAfterCurrentScreen == 0;
 
+        final FormEntryCaption repeatCaptionPrompt = FormEntryActivity.mFormController.getCaptionPrompt();
+
         // Assign title and text strings based on the current state
         String backText = Localization.get("repeat.dialog.go.back");
-        String addAnotherText = Localization.get("repeat.dialog.add");
+
+        boolean hasRepetitions = FormEntryActivity.mFormController.getLastRepeatCount() > 0;
+        String addAnotherText = repeatCaptionPrompt.getRepeatText(hasRepetitions ? "add" : "add-empty");
+
         String title, skipText;
+
         if (!nextExitsForm) {
             skipText = Localization.get("repeat.dialog.leave");
         } else {
             skipText = Localization.get("repeat.dialog.exit");
         }
-        if (FormEntryActivity.mFormController.getLastRepeatCount() > 0) {
-            title = Localization.get("repeat.dialog.add.another", FormEntryActivity.mFormController.getLastGroupText());
-        } else {
-            title = Localization.get("repeat.dialog.add.new", FormEntryActivity.mFormController.getLastGroupText());
-        }
+
+        title = addAnotherText;
```

Comment:
nit: can combine assignment with declaration here.

Reply:
We don't even need that variable hence removed it now.